libtransmission/announce-list.cc

```diff
@@ -0,0 +1,259 @@
+/*
+ * This file Copyright (C) Mnemosyne LLC
+ *
+ * It may be used under the GNU GPL versions 2 or 3
+ * or any future license endorsed by Mnemosyne LLC.
+ *
+ */
+
+#include <algorithm>
+#include <set>
+#include <string>
+#include <string_view>
+
+#include "transmission.h"
+
+#include "announce-list.h"
+#include "metainfo.h"
+#include "utils.h"
+#include "variant.h"
+
+size_t tr_announce_list::set(char const* const* announce_urls, tr_tracker_tier_t const* tiers, size_t n)
```

Comment:
Is `announce_urls` a double pointer or you actually meant `char const* const announce_urls`?

Reply:
It's a double pointer; that is, an array of zero-terminated C strings. `libtransmission/transmission.h` needs to stay both C++ and C friendly.

(Small disclaimer to above since that comment in this PR might raise an eyebrow: yes, this PR does break C compatibility in transmission.h by putting a std::shared_ptr in tr_info. That will be fixed shortly, either in 2285 or a followup.

...hm, it might make sense to make `tr_announce_list_set` take a span of string_views, and have `tr_torrentSetAnnounceList()` take the char const* const* and mutate it to a span of  string views before it calls announce_list.set()

Reply:
To be honest with you, I have never seen such code; only code like `char const* const foo;`, but not like the aforementioned example.